tests/app/main/views/test_template_folders.py

```diff
@@ -0,0 +1,166 @@
+import uuid
+
+import pytest
+from flask import url_for
+
+from tests.conftest import SERVICE_ONE_ID, normalize_spaces
+
+PARENT_FOLDER_ID = '7e979e79-d970-43a5-ac69-b625a8d147b0'
+CHILD_FOLDER_ID = '92ee1ee0-e4ee-4dcc-b1a7-a5da9ebcfa2b'
+
+
+@pytest.mark.parametrize('parent_folder_id', [None, PARENT_FOLDER_ID])
+def test_add_page_shows_option_for_folder(
+    client_request,
+    service_one,
+    parent_folder_id,
+    mocker,
+    mock_get_service_templates,
+    mock_get_organisations_and_services_for_user,
+):
+    service_one['permissions'] += ['edit_folders']
+    mocker.patch('app.service_api_client.get_service', return_value={"data": service_one})
+
+    page = client_request.get(
+        'main.add_template_by_type',
+        service_id=service_one['id'],
+        template_folder_id=parent_folder_id,
+        _test_page_title=False
+    )
+
+    radios = page.select('input[type=radio]')
+    labels = page.select('label')
+
+    assert [x['value'] for x in radios] == ['email', 'sms', 'copy-existing', 'folder']
+    assert [x.text.strip() for x in labels] == [
+        'Email template',
+        'Text message template',
+        'Copy of an existing template',
+        'Folder'
+    ]
+
+
+@pytest.mark.parametrize('parent_folder_id', [None, PARENT_FOLDER_ID])
+def test_get_add_template_folder_page(client_request, service_one, parent_folder_id):
+    service_one['permissions'] += ['edit_folders']
+
+    page = client_request.get(
+        'main.add_template_folder',
+        service_id=service_one['id'],
+        template_folder_id=parent_folder_id
+    )
+
+    assert page.find('input', attrs={'name': 'name'}) is not None
```

Comment:
Prefer using CSS selectors:
```suggestion
    assert page.select_one('input[name=name]') is not None
```